### PR TITLE
Add support for Petrified Blood EHP

### LINKED ISSUE #2737: Add support of eHP for petrified blood
Currently the eHP and maximum hit taken is calculated without the 40% damage mitigation

If we can have this included to have an accurate representation for the one shot potential.

Tnx

### PR DESCRIPTION
**Pull request by deathbeam**
This adds support for Petrified Blood life loss prevention to EHP calcs
in similar fashion as Mind over Matter. The prevention gain is checked
against being at low life or not, below low life it will use full value
(e.g 40 base) above full life it will use halved value (Pathfinders for
example utilize this mechanic and recover to full only via flasks).

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

Fixes #2737

### Link to a build that showcases this PR:

https://pastebin.com/TxUzDmsC (full life Petrified Blood Pathfinder)

### Before screenshot:

image

### After screenshot:

image

### CODE REVIEW

File: src/Data/Skills/act_str.lua [hunk 1]
```diff
@@ -4929,6 +4929,13 @@ skills["PetrifiedBlood"] = {
 		["skill_grants_life_cost_%_mana_cost_while_not_on_low_life"] = {
 			mod("ManaCostAsLifeCost", "BASE", nil, 0, 0, { type = "GlobalEffect", effectType = "Buff", unscalable = true }, { type="Condition", var="LowLife", neg = true }),
 		},
+		["petrified_blood_%_life_loss_below_half_from_hit_to_prevent"] = {
+			mod("LifeLossBelowHalfPrevented", "BASE", nil, 0, 0, { type = "GlobalEffect", effectType = "Buff" }),
+		},
+		["cannot_recover_above_low_life_except_flasks"] = {
+			flag("CannotLeechLife", { type = "GlobalEffect", effectType = "Buff" }, { type="Condition", var="LowLife", neg = true }),
+			flag("NoLifeRegen", { type = "GlobalEffect", effectType = "Buff" }, { type="Condition", var="LowLife", neg = true }),
```
[the remark this answers is not in the document]
Comment (deathbeam) [reply src/Data/Skills/act_str.lua]: Hmm you are right and i also dont rly want to deal with that, my main goal was the EHP anyway, removed the regen part

File: src/Modules/CalcSections.lua [hunk 2]
```diff
@@ -1601,6 +1601,28 @@ return {
 			{ modName = { "GuardAbsorbRate", "ChaosGuardAbsorbRate", "GuardAbsorbLimit", "ChaosGuardAbsorbLimit" } },
 		},
 	},
+	{ label = "Prevented life loss", haveOutput = "AnyPreventedLifeLoss",
```
Comment (Nostrademous) [inline src/Modules/CalcSections.lua:1604]: can you just capitalize this to `Prevented Life Loss` to keep format with other rows in that table please?

Comment (deathbeam) [reply src/Modules/CalcSections.lua]: Sure, updated

**Comment by deathbeam** [comment]
Updated the PR and moved Petrified Blood calcuations before guard skill, as based on this post that is the correct order: https://www.pathofexile.com/forum/view-thread/3123794

The actual order btw is (from bottom to top):

Petrified blood, x before life (so mind over matter), energy shield, guard, aegis, frost shield, jinxed juju

So I think also mind over matter is in incorrect position and should be *before* guard skill, but I am not going to fix that here as that is different issue.

**Comment by Regisle** [comment]
@deathbeam the whole thing is being reworked anyway in #2390 and its order is correctly applied there, also your list is missing ward

**Comment by deathbeam** [comment]
Ah i see. And yea that post is missing info about ward and i was just going based on that.

**Comment by Wires77** [comment]
@deathbeam The calc defences rework PR has been merged into dev, so can you take a look at the conflicts for me and we'll take another look

**Comment by Regisle** [comment]
I am in the process of fixing it to work with the new system, as well as giving people an idea of what "degen" they can expect

**Comment by deathbeam** [comment]
Closing in favour of #4060 because of the calc defenses rework

**Pull request closed without merging**